src/GuessControl.js

```diff
@@ -1,47 +1,34 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
 import Button from "./Button";
 
-class GuessControl extends Component {
-  constructor(props) {
-    super(props);
+// Functional Component
+function GuessControl({ onGuess }) {
+  const [currentGuess, setCurrentGuess] = useState('')
 
-    this.state = {
-      currentGuess: "",
-    };
-
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.onSubmitGuess = this.onSubmitGuess.bind(this);
+  const handleInputChange = (event) => {
+    setCurrentGuess(event.target.value)
   }
+  const onSubmitGuess = () => {
+    onGuess(currentGuess)
```

Comment:
The instructions stated to convert the `currentGuess` variable to a number. To do this, we can use the `parseInt()` function that comes with JavaScript.